services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/northbound/links/LinkSpec.groovy

```diff
@@ -0,0 +1,177 @@
+        [flow1, flow2].each { assert it in linkFlows }
+        [flow3, flow4].each { assert !(it in linkFlows) }
+
+        when: "Bring all ports down on source switch that are involved in current and alternative paths"
+        List<List<PathNode>> possibleFlowPaths = db.getPaths(srcSwitch.dpId, dstSwitch.dpId)*.path
+        List<PathNode> broughtDownPorts = []
+        possibleFlowPaths.unique { it.first() }.each { path ->
+            def src = path.first()
+            broughtDownPorts.add(src)
+            northboundService.portDown(src.switchId, src.portNo)
+        }
+
+        then: "All flows go to 'Down' status"
+        Wrappers.wait(rerouteDelay + WAIT_OFFSET) {
+            [flow1, flow2, flow3, flow4].each { assert northboundService.getFlowStatus(it.id).status == FlowState.DOWN }
+        }
+
+        when: "Get all flows going through the link from source switch to 'internal' switch"
+        linkFlows = northboundService.getLinkFlows(islToInternal.srcSwitch.dpId, islToInternal.srcPort,
+                islToInternal.dstSwitch.dpId, islToInternal.dstPort)
+
+        then: "All created flows are in the response list"
+        [flow1, flow2, flow3, flow4].each { assert it in linkFlows }
+
+        when: "Get all flows going through the link from 'internal' switch to destination switch"
+        linkFlows = northboundService.getLinkFlows(islFromInternal.srcSwitch.dpId, islFromInternal.srcPort,
+                islFromInternal.dstSwitch.dpId, islFromInternal.dstPort)
+
+        then: "Only the first and second flows are in the response list"
+        [flow1, flow2].each { assert it in linkFlows }
+        [flow3, flow4].each { assert !(it in linkFlows) }
+
+        when: "Bring ports up"
+        broughtDownPorts.each { northboundService.portUp(it.switchId, it.portNo) }
+
+        then: "All flows go to 'Up' status"
+        Wrappers.wait(rerouteDelay + discoveryInterval + WAIT_OFFSET) {
+            [flow1, flow2, flow3, flow4].each {
+                assert northboundService.getFlowStatus(flow1.id).status == FlowState.UP
+            }
+        }
+
+        and: "Delete all created flows"
+        [flow1, flow2, flow3, flow4].each { assert northboundService.deleteFlow(it.id) }
+    }
+
+    @Unroll
+    def "Unable to get flows for NOT existing link (#item doesn't exist) "() {
+        when: "Get flows for NOT existing link"
+        northboundService.getLinkFlows(srcSwId, srcSwPort, dstSwId, dstSwPort)
+
+        then: "An error is received (404 code)"
+        def exc = thrown(HttpClientErrorException)
+        exc.rawStatusCode == 404
+        exc.responseBodyAsString.to(MessageError).errorMessage ==
+                "There is no ISL between $srcSwId-$srcSwPort and $dstSwId-$dstSwPort."
+
+        where:
+        srcSwId                   | srcSwPort        | dstSwId                   | dstSwPort        | item
+        new SwitchId("123456789") | getIsl().srcPort | getIsl().dstSwitch.dpId   | getIsl().dstPort | "src_switch"
```

Comment:
Does it even allow you to create such SwitchId? I'd rather went with at least valid 'format' of the id, sth like `00:00:00:00:00:00:00:00`

Reply:
Yes, it does. But then it's converted to 00:00:00:01:23:45:67:89 somehow.